docs/content/docs/connectors/datastream/prometheus.md

```diff
@@ -0,0 +1,473 @@
+are written in the same order as they are received by the sub-task. 
+
+To ensure all records belonging to the same time-series (i.e. `PrometheusTimeSeries` with identical list of `Label` and `metricName`)
+are written by the same sink subtask, the stream of `PrometheusTimeSeries` must be keyed using `PrometheusTimeSeriesLabelsAndMetricNameKeySelector`.
+
+```java
+DataStream<MyRecord> inputRecords;
+// ...
+KeyedStream<PrometheusTimeSeries> timeSeries = inputRecords
+        .map(new MyRecordToTimeSeriesMapper())
+        .keyBy(new PrometheusTimeSeriesLabelsAndMetricNameKeySelector());
+
+timeSeries.sinkTo(prometheusSink);
+```
+
+Using this key selector prevents accidental out-of-orderness due to repartitioning before the sink operator.
+However, the user is responsible to retain ordering to this point by partitioning the records correctly.
+
+
+## Error handling
+
+This paragraph covers handling of errors conditions when writing data to the Remote Write endpoint. 
+
+There are four types of error conditions:
+
+1. Retryable errors due to temporary error conditions in the Remote-Write server or due to throttling: `5xx` or `429` http responses, connectivity issues.
+2. Non-retryable errors due to data violating any of the constraints, malformed data or out-of-order samples: `4xx` http responses, except `429`, `403` and `404`.
+3. Fatal error response: authentication failures (`403` http response) or incorrect endpoint path (`404` http response).
+4. Any other unexpected failure while writing, due to exceptions while writing to the Prometheus endpoint.
+
+
+### On-error behaviors
+
+When any of the above error is encountered, the connector implements one of these two behaviors:
+
+1. `FAIL`: throw an unhandled exception, the job fails
+2. `DISCARD_AND_CONTINUE`: discard the request that caused the error, and continue with the next record.
+
+When a write request is discarded on `DISCARD_AND_CONTINUE`, all the following happens:
+
+1. Log a message at `WARN` level with the cause of the error. When the error is caused by a response from the endpoint, the payload of the response from the endpoint is included.
+2. Increase [counter](#connector-metrics) metrics, to count the number of rejected samples and write requests.
+3. **Drop the entire write request**. Note that due to [batching](#batching), a write request may contain multiple `PrometheusTimeSeries`.
+4. Continue with the next input record.
+
+
+{{< hint info >}}Prometheus Remote Write does not support partial failures. 
+Due to [batching](#batching), a single write request may contain multiple input records (`PrometheusTimeSeries`). 
+If a request contains even a single offending record, the entire write request (the entire batch) must be discarded.{{< /hint >}}
+
+
+#### Retryable error responses
+
+A typical retryable error condition is enpoint throttling, with a `429, Too Many Requests` response.
```

Comment:
We might want to use US english!

Reply:
We are using US English. Beside the misspelled "endpoint" that I will fix, what not US English here?